homeassistant/components/renault/services.py

```diff
@@ -0,0 +1,194 @@
+"""Support for Renault services."""
+from __future__ import annotations
+
+from datetime import datetime
+import logging
+from types import MappingProxyType
+from typing import Any
+
+import voluptuous as vol
+
+from homeassistant.core import HomeAssistant, ServiceCall
+from homeassistant.helpers import config_validation as cv
+
+from .const import DOMAIN
+from .renault_hub import RenaultHub
+from .renault_vehicle import RenaultVehicleProxy
+
+LOGGER = logging.getLogger(__name__)
+
+ATTR_CHARGE_MODE = "charge_mode"
+ATTR_SCHEDULES = "schedules"
+ATTR_TEMPERATURE = "temperature"
+ATTR_VIN = "vin"
+ATTR_WHEN = "when"
+
+REGEX_VIN = "(?i)^VF1[\\w]{14}$"
+
+SERVICE_AC_CANCEL_SCHEMA = vol.Schema(
+    {
+        vol.Required(ATTR_VIN): cv.matches_regex(REGEX_VIN),
+    }
+)
+SERVICE_AC_START_SCHEMA = vol.Schema(
+    {
+        vol.Required(ATTR_VIN): cv.matches_regex(REGEX_VIN),
+        vol.Required(ATTR_TEMPERATURE): cv.positive_float,
+        vol.Optional(ATTR_WHEN): cv.datetime,
+    }
+)
+SERVICE_CHARGE_SET_MODE_SCHEMA = vol.Schema(
+    {
+        vol.Required(ATTR_VIN): cv.matches_regex(REGEX_VIN),
+        vol.Required(ATTR_CHARGE_MODE): vol.In(
+            ["always", "always_charging", "schedule_mode"]
```

Comment:
Consider making this a select entity in a select platform. If this is the service the user most frequently uses for changing charging it may be more convenient to access it as an entity.

Reply:
I've created PR #55494 but I am unsure if this is a new feature, or if it becomes a breaking change.